### PR TITLE
mrc-4260 dockerise api

### PR DESCRIPTION
**Pull request by hillalex**
Do the very simplest thing to dockerise the app - use the built in `bootBuildImage` gradle task to create a docker image on CI, then smoke test and push the image.

### CODE REVIEW

File: api/README.md [hunk 1]
```diff
@@ -20,3 +20,8 @@ Execute tests on the command line or through IntelliJ
 To run a specific test alone, add `--test` + the \
 [fully qualified class name](https://docs.gradle.org/current/userguide/java_testing.html#full_qualified_name_pattern)\
 to the command. For example, the command for running AppTest.kt would be: `./api/gradlew -p api/app :app:test --tests AppTest`
+
+## Building a docker image
+1. `./api/scripts/build` builds a docker image
+2. `./api/scripts/build-and-push` builds and pushes an image to dockerhub and runs on CI
```
Comment (EmmaLRussell) [inline api/README.md:26]: Not sure what "..and runs on CI" means here - the last thing the script does does is push the image

**Review by LekanAnni — APPROVED** [review]
Works like magic 👌

**Review by EmmaLRussell — COMMENTED** [review]
Is it ok for the smoke test to be getting a 404?
https://github.com/mrc-ide/packit/actions/runs/5003712478/jobs/8965289792?pr=16#step:11:58

File: api/README.md (hunk 1 above)
Comment (hillalex) [reply api/README.md:26]: Ha, I meant that this script is run on CI! Will rephrase :)

**Comment by hillalex** [comment]
> Is it ok for the smoke test to be getting a 404? https://github.com/mrc-ide/packit/actions/runs/5003712478/jobs/8965289792?pr=16#step:11:58

Hmm, well I guess technically it means the app is running, but no not ideal. Will make it more precise!

**Review by EmmaLRussell — APPROVED** [review]

**Pull request merged**